### PR TITLE
Support getting function name & path params details from environment API

### LINKED ISSUE #39783: [Improvement]: Add support to get function name, path params details from environment API
### Description

When we deal with native functions, sometimes we need to assign a generic java function to multiple ballerina functions, and the execution logic in the native function needs to know the function name and other path param details to execute the logic. There is no way to get this information in the native function.

### Describe your problem(s)

In the persistent implementation, we generate a Ballerina client and resource functions. these are external resource functions and the native java function is a generic function that can attach to multiple generated resource functions. when we are executing the native code. we need to know the resource function name and path parameters.

### Describe your solution(s)

Give a new API from the Environment class to get the function name and path params.

### Related area

-> Runtime

### Related issue(s) (optional)

https://github.com/ballerina-platform/ballerina-lang/issues/39737

### Suggested label(s) (optional)

jballerina

### Suggested assignee(s) (optional)

@warunalakshitha

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #39783

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java [hunk 1]
```diff
@@ -45,6 +48,20 @@ public Environment(Strand strand, Module currentModule) {
...
+        this(strand, currentModule);
+        this.funcName = funcName;
+        this.funcPathParams = funcPathParams;
+    }
+
+    public String getFunctionName() {
+        return funcName;
+    }
+
+    public Parameter[] getFunctionPathParameters() {
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java:61]:
```suggestion
    public Parameter[] getPathParameters() {
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java [hunk 2]
```diff
@@ -34,6 +35,8 @@ public class Environment {
     private final Strand strand;
     private Future future;
     private Module currentModule;
+    private String funcName;
+    private Parameter[] funcPathParams;
```
Comment (warunalakshitha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Environment.java:39]:
```suggestion
    private Parameter[] pathParams;
```

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java [hunk 3]
```diff
@@ -604,8 +611,11 @@ private void genJICallTerm(JIMethodCall callIns, int localVarOffset) {
             this.mv.visitVarInsn(ALOAD, localVarOffset); // load the strand
             // load the current Module
             mv.visitFieldInsn(GETSTATIC, this.moduleInitClass, CURRENT_MODULE_VAR_NAME, GET_MODULE);
+            // load function name
+            mv.visitLdcInsn(func.name.getValue());
+            loadFuncPathParams(mv, (BInvokableTypeSymbol) func.type.tsymbol);
             mv.visitMethodInsn(INVOKESPECIAL, BAL_ENV, JVM_INIT_METHOD,
-                               INIT_BAL_ENV, false);
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java:608]: Can we remove this constant?

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java:608]: This is used in `generatePlatformIns` in `JVMInstructionGen`

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java [hunk 4]
```diff
@@ -649,6 +659,46 @@ private void genJICallTerm(JIMethodCall callIns, int localVarOffset) {
...
+                    break;
+                }
+                params.add(param);
+            }
+        }
+
+        mv.visitLdcInsn((long) params.size());
+        mv.visitInsn(L2I);
+        mv.visitTypeInsn(ANEWARRAY, FUNCTION_PARAMETER);
+        for (int i = 0; i < params.size(); i++) {
```
Comment (warunalakshitha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java:677]: We already have loadFunctionParameters in jvmtypegen. can we reuse those logic?

Comment (Nadeeshan96) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTerminatorGen.java:677]: Yes, I will refactor to remove duplicate parts

**Comment by warunalakshitha** [comment]
Lets write some tests with separate PR.

**Comment by Nadeeshan96** [comment]
Tests are added with https://github.com/ballerina-platform/ballerina-lang/pull/39843.

**Pull request merged**